lib/device/device.dart

```diff
@@ -51,172 +44,53 @@ class DeviceScreen extends Screen {
         ])
     ]);
 
-    _rebuildTogglesDiv();
-  }
-
-  void _handleConnectionStart(VmServiceWrapper service) {
-    extensionTracker = new ExtensionTracker(service);
-    extensionTracker.start();
-
-    extensionTracker.onChange.listen((_) {
-      framesChartStateMixin.setState(() {
-        if (extensionTracker.hasIsolateTargets && !visible) {
-          visible = true;
-        }
-
-        _rebuildTogglesDiv();
-      });
-    });
-
-    deviceStatus.element.text =
-        '${serviceManager.vm.targetCPU} ${serviceManager.vm.architectureBits}-bit';
-  }
-
-  void _handleConnectionStop(dynamic event) {
-    extensionTracker?.stop();
-
-    deviceStatus.element.text = '';
+    _buildTogglesDiv();
   }
 
   @override
   HelpInfo get helpInfo => null;
 
-  void _rebuildTogglesDiv() {
-    if (togglesDiv == null || extensionTracker == null) {
+  void _buildTogglesDiv() {
+    if (togglesDiv == null) {
       return;
     }
 
-    togglesDiv.clear();
-
-    _createBoolToggle('ext.flutter.debugPaint');
-    _createBoolToggle('ext.flutter.debugPaintBaselinesEnabled');
-    _createBoolToggle('ext.flutter.repaintRainbow');
-    _createBoolToggle('ext.flutter.showPerformanceOverlay');
-    _createBoolToggle('ext.flutter.debugAllowBanner');
-    _createBoolToggle('ext.flutter.profileWidgetBuilds');
+    _createBoolToggle(extensions.debugPaint);
+    _createBoolToggle(extensions.debugPaintBaselines);
+    _createBoolToggle(extensions.repaintRainbow);
+    _createBoolToggle(extensions.performanceOverlay);
+    _createBoolToggle(extensions.debugAllowBanner);
+    _createBoolToggle(extensions.profileWidgetBuilds);
   }
 
   void _createBoolToggle(String rpc) {
```

Comment:
consider moving this to be a helper method in a utils library that creates a CoreElement containing a bool toggle. This is functionality we need on a bunch of pages.

Reply:
Created a checkbox CoreElement helper as well as a createExtensionCheckbox helper.